includes/functions.php

```diff
@@ -158,31 +168,34 @@ function getImageName($device, $use_database = true)
     $device['os'] = strtolower($device['os']);
 
     // fetch from the database
-    if ($use_database && !empty($device['icon']) && file_exists($config['html_dir'] . "/images/os/" . $device['icon'] . ".png")) {
+    if ($use_database && !empty($device['icon']) && file_exists($config['html_dir'] . '/images/os/' . $device['icon'])) {
         return $device['icon'];
     }
 
     // linux specific handling, distro icons
+    $distro = null;
     if ($device['os'] == "linux") {
         $features = strtolower(trim($device['features']));
         list($distro) = explode(" ", $features);
-        if (file_exists($config['html_dir'] . "/images/os/$distro" . ".png")) {
-            return $distro;
-        }
     }
 
-    // use the icon from os config
-    if (!empty($config['os'][$device['os']]['icon']) && file_exists($config['html_dir'] . "/images/os/" . $config['os'][$device['os']]['icon'] . ".png")) {
-        return $config['os'][$device['os']]['icon'];
-    }
+    $possibilities = array(
+        $distro,
+        $config['os'][$device['os']]['icon'],
+        $device['os'],
+    );
 
-    // guess the icon has the same name as the os
-    if (file_exists($config['html_dir'] . '/images/os/' . $device['os'] . '.png')) {
-        return $device['os'];
+    foreach ($possibilities as $icon) {
+        if (!empty($icon)) {
```

Comment:
You don't need this check if you use is_file() instead of file_exists() (which returns true for directories)